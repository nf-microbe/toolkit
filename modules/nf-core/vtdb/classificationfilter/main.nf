process VTDB_CLASSIFICATIONFILTER {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(combined_data)
    path classification_filters

    output:
    tuple val(meta), path("${prefix}_class_data.tsv")   , emit: class_data
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args        = task.ext.args ?: ''
    prefix          = task.ext.prefix ?: "${meta.id}"
    """
    classificationfilter.py \\
        --input ${combined_data} \\
        --filters ${classification_filters} \\
        --output ${prefix}_class_data.tsv \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        numpy: \$(python -c "import numpy; print(numpy.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_class_data.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        numpy: \$(python -c "import numpy; print(numpy.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """
}
