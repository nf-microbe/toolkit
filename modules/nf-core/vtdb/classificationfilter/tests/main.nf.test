nextflow_process {

    name "Test Process: VTDB_CLASSIFICATIONFILTER"
    script "../main.nf"
    process "VTDB_CLASSIFICATIONFILTER"

    tag "modules_nfcore"
    tag "modules"
    tag "vtdb"
    tag "vtdb/classificationfilter"

    test("combined.tsv + classification_filters.tsv") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/classificationfilter/test_combined_data.tsv", checkIfExists: true)
                ]
                input[1] = file("${projectDir}/assets/filters/classification_filters.tsv", checkIfExists: true)
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/classificationfilter/test_combined_data.tsv", checkIfExists: true)
                ]
                input[1] = file("${projectDir}/assets/filters/classification_filters.tsv", checkIfExists: true)
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
