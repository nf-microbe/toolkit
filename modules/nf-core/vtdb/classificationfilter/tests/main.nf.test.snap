{
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_class_data.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6b20fd0082e7a3a64693561230f4d546"
                ],
                "class_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_class_data.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b20fd0082e7a3a64693561230f4d546"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T08:32:22.340763392"
    },
    "combined.tsv + classification_filters.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_class_data.tsv:md5,e89662c598aab651443a85a22300d164"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6b20fd0082e7a3a64693561230f4d546"
                ],
                "class_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_class_data.tsv:md5,e89662c598aab651443a85a22300d164"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b20fd0082e7a3a64693561230f4d546"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-16T14:41:32.479052871"
    }
}